include/lsst/ip/isr/isr.h

```diff
@@ -0,0 +1,217 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the LSST License Statement and 
+ * the GNU General Public License along with this program.  If not, 
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+ 
+/**
+  * \file
+  *
+  * \ingroup isr
+  *
+  * \brief Implementation of the templated Instrument Signature Removal 
+  * stage of the nightly LSST Image Processing Pipeline.
+  *
+  * \author Nicole M. Silvestri / ACB, University of Washington
+  *
+  * Contact: nms@astro.washington.edu
+  *
+  * \version
+  *
+  * LSST Legalese here...
+  */
+	
+#ifndef LSST_IP_ISR_ISR_H
+#define LSST_IP_ISR_ISR_H
+	
+#include <memory>
+#include <string>
+#include <vector>
+#include <cmath>
+
+#include "lsst/afw/math.h"
+#include "lsst/afw/math/Statistics.h"
+#include "lsst/afw/image.h"
+#include "lsst/pex/exceptions/Exception.h"
+
+/** \brief Remove all non-astronomical counts from the Chunk Exposure's pixels.
+  * 
+  */
+	
+namespace lsst {
+namespace ip {
+namespace isr {
+
+    /** Signature strings associated with each stage of the ISR
+     * 
+     * @note Added to Exposure metadata after stage processing
+     *
+     * @note As implementation detail, no more than 8 characters for fits
+     * compliance?
+     */
+    std::string const& ISR_LIN     = "ISR_LIN";    ///< Linearization
+    std::string const& ISR_OSCAN   = "ISR_OSCAN";  ///< Overscan
+    std::string const& ISR_TRIM    = "ISR_TRIM";   ///< Trim
+    std::string const& ISR_BIAS    = "ISR_BIAS";   ///< Bias 
+    std::string const& ISR_DFLAT   = "ISR_DFLAT";  ///< Dome flat
+    std::string const& ISR_ILLUM   = "ISR_ILLUM";  ///< Illumination correction
+    std::string const& ISR_BADP    = "ISR_BADP";   ///< Bad pixel mask
+    std::string const& ISR_SAT     = "ISR_SAT";    ///< Saturated pixels
+    std::string const& ISR_FRING   = "ISR_FRING";  ///< Fringe correction
+    std::string const& ISR_DARK    = "ISR_DARK";   ///< Dark correction
+    std::string const& ISR_PUPIL   = "ISR_PUPIL";  ///< Pupil correction
+    std::string const& ISR_CRREJ   = "ISR_CRREJ";  ///< Cosmic ray rejection
+    std::string const& ISR_BACKSUB = "ISR_BACKSUB";  ///< Background subtraction
+
+    enum StageId {
+        ISR_LINid   = 0x1,   ///< Linearization
+        ISR_OSCANid = 0x2,   ///< Overscan
+        ISR_TRIMid  = 0x4,   ///< Trim
+        ISR_BIASid  = 0x8,   ///< Bias 
+        ISR_DFLATid = 0x10,  ///< Dome flat
+        ISR_ILLUMid = 0x20,  ///< Illumination correction
+        ISR_BADPid  = 0x40,  ///< Bad pixel mask
+        ISR_SATid   = 0x80,  ///< Saturated pixels
+        ISR_FRINid  = 0x100, ///< Fringe correction
+        ISR_DARKid  = 0x200, ///< Dark correction
+        ISR_PUPILid = 0x400, ///< Pupil correction
+        ISR_CRREJid = 0x800, ///< Cosmic ray rejection
+        ISR_BACKSUBid = 0x1000, ///< Cosmic ray rejection
+    };
```

Comment:
Can we remove all the above (`StageId` and the "signature strings")?

(I realise this is more than what you originally signed up for, so feel free to tell me to take a hike.)